app/javascript/containers/ReviewModeContainer.jsx

```diff
@@ -108,55 +71,97 @@ class ReviewModeContainer extends React.Component {
     for (let category in this.props.categories) {
       let stateCategory = this.props.categories[category];
       count += 1;
-      categoryMap[stateCategory.id] = getQuestionsByCategory(stateCategory.id, questions);
-      stack.push(<CategoryHeader
-        category={stateCategory}
-        number={count}
-        buildingId={building.id}
-      />);
+      categoryMap[stateCategory.id] = getQuestionsByCategory(
+        stateCategory.id,
+        questions
+      );
+      stack.push(
+        <CategoryHeader
+          category={stateCategory}
+          number={count}
+          buildingId={building.id}
+        />
+      );
       stack = stack.concat(
         <table cellSpacing="0">
           <tbody>
-          {mapCategorytoQuestions(categoryMap, stateCategory.id, building)}
+            {mapCategorytoQuestions(categoryMap, stateCategory.id, building)}
           </tbody>
         </table>
       );
     }
     return stack;
   }
 
-
   render() {
+    var parentQuestionsForDelegations = this.props.questions.filter(
```

Comment:
Can you actually move this code for finding delegations and all their dependent questions to a separate function, and call it inside of `submitDelegations` instead of `render`? I'm worried that `render` will get called any time a prop updates, which would cause us to repeat all this fancy question finding logic unnecessarily instead of only running it once when the Submit button is clicked

Reply:
omg I knew there was something sketchy about this LOL I should've realized it!! good call